src/main/java/school/hei/haapi/service/GradeService.java

```
@@ -70,4 +72,12 @@ public List<Grade> getParticipantsGradeForExam(
             ? Pageable.unpaged()
             : PageRequest.of((page.getValue() - 1), pageSize.getValue()));
   }
+
+  private OptionalDouble getExamAverageGrade(String examId) {
+    return gradeDao.getGradesByExamId(examId).stream().mapToDouble(Grade::getScore).average();
+  }
+
+  public ExamGradeStats getExamGradeStats(String examId) {
+    return new ExamGradeStats().average(getExamAverageGrade(examId).orElse(0));
```

should throw not found if cannot find stat, not 0